projects/igniteui-angular/src/lib/paginator/paginator.component.ts

```diff
@@ -0,0 +1,273 @@
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import {
+    Component,
+    Input,
+    Output,
+    OnInit,
+    OnChanges,
+    NgModule,
+    Optional,
+    Inject,
+    EventEmitter,
+    HostBinding,
+    SimpleChanges
+} from '@angular/core';
+import { CurrentResourceStrings } from '../core/i18n/resources';
+import { IDisplayDensityOptions, DisplayDensityToken, DisplayDensityBase, DisplayDensity } from '../core/displayDensity';
+import { IgxSelectModule } from '../select/index';
+import { IgxIconModule } from '../icon/index';
+import { IgxButtonModule } from '../directives/button/button.directive';
+import { IgxRippleModule } from '../directives/ripple/ripple.directive';
+
+
+@Component({
+    selector: 'igx-paginator',
+    templateUrl: 'paginator.component.html',
+})
+export class IgxPaginatorComponent extends DisplayDensityBase implements OnInit, OnChanges {
+    /**
+     * @hidden
+     */
+    public totalPages: number;
+
+    /**
+     * @hidden
+     */
+    @HostBinding('class') get classes(): string {
+        switch (this.displayDensity) {
+            case DisplayDensity.cosy:
+                return 'igx-grid-paginator--cosy';
+            case DisplayDensity.compact:
+                return 'igx-grid-paginator--compact';
+            default:
+                return 'igx-grid-paginator';
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    @Input()
+    public page = 0;
```

Comment:
IMO it would be better to expose `page` and `perPage` as two-way bind able properties.
You can check [this](https://angular.io/guide/template-syntax#two-way-binding-) and you already implemented it for the `page` property.

This way you won't need the onInit and onChanges hooks and you can remove the `handleChange()` method.